be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -674,129 +664,359 @@ Status JsonReader::_read_and_parse_json() {
-        _counter->num_rows_filtered++;
         return Status::DataQualityError(err_msg.c_str());
     }
 
     _doc_stream_itr = _doc_stream.begin();
+    _stream_is_empty = false;
 
     return Status::OK();
 }
 
 // _construct_column constructs column based on no value.
-void JsonReader::_construct_column(simdjson::ondemand::value& value, Column* column, const TypeDescriptor& type_desc) {
+Status JsonReader::_construct_column(simdjson::ondemand::value& value, Column* column,
+                                     const TypeDescriptor& type_desc) {
     simdjson::ondemand::json_type tp;
     auto err = value.type().get(tp);
     if (err) {
-        return;
+        std::string err_msg = strings::Substitute("Failed to determine the type of column value. code=$0, error=$1",
+                                                  err, simdjson::error_message(err));
+        return Status::DataQualityError(err_msg.c_str());
     }
 
     switch (tp) {
     case simdjson::ondemand::json_type::null: {
         column->append_nulls(1);
-        break;
+        return Status::OK();
     }
 
     case simdjson::ondemand::json_type::boolean: {
         bool ok;
         auto err = value.get_bool().get(ok);
         if (UNLIKELY(err)) {
-            column->append_nulls(1);
-            break;
+            std::string err_msg = strings::Substitute("Failed to parse the column value as bool. code=$0, error=$1",
+                                                      err, simdjson::error_message(err));
+            return Status::DataQualityError(err_msg.c_str());
         }
 
         if (ok) {
-            column->append_strings(literal_1_slice_vector);
+            _construct_string_column(column, literal_1_slice);
         } else {
-            column->append_strings(literal_0_slice_vector);
+            _construct_string_column(column, literal_0_slice);
         }
-        break;
+        return Status::OK();
     }
 
     case simdjson::ondemand::json_type::number: {
-        simdjson::ondemand::number_type tp;
+        return _construct_column_with_numeric_value(value, column, type_desc);
+    }
+
+    case simdjson::ondemand::json_type::string: {
+        return _construct_column_with_string_value(value, column, type_desc);
+    }
 
-        auto err = value.get_number_type().get(tp);
+    case simdjson::ondemand::json_type::object:
+    case simdjson::ondemand::json_type::array: {
+        std::unique_ptr<char[]> buf;
+        size_t buflen{};
+        JsonFunctions::minify_json_to_string(value, buf, buflen);
+
+        _construct_string_column(column, Slice{buf.get(), buflen});
+        return Status::OK();
+    }
+    }
+    return Status::OK();
+}
+
+Status JsonReader::_construct_column_with_numeric_value(simdjson::ondemand::value& value, Column* column,
+                                                        const TypeDescriptor& type_desc) {
+    simdjson::ondemand::number_type tp;
+
+    auto err = value.get_number_type().get(tp);
```

Comment:
If this is a null value and the corresponding column in StarRocks is nullable, a null record should be appended into the column, instead of return an error.